src/vs/workbench/contrib/terminalContrib/find/browser/terminalFindWidget.ts

```diff
@@ -100,9 +100,29 @@ export class TerminalFindWidget extends SimpleFindWidget {
 			}
 		}));
 
+		this._setupSearchEventListeners();
 		this.updateResultCount();
 	}
 
+	private _setupSearchEventListeners(): void {
+		const xterm = this._instance.xterm;
+		if (!xterm) {
+			return;
+		}
+
+		// Disable copy-on-selection during search to prevent search result from overriding clipboard
+		this._register(xterm.onBeforeSearch(() => {
+			this._overrideCopyOnSelectionDisposable?.dispose();
+			this._overrideCopyOnSelectionDisposable = TerminalClipboardContribution.get(this._instance)?.overrideCopyOnSelection(false);
```

Comment:
Let's make this a MutableDisposable while we're here, it'll make things simpler.